tests/nix_tests/test_patches.py

```diff
@@ -0,0 +1,337 @@
+                        "coins": "0aevmos",
+                    }
+                ],
+            },
+            "vesting": {
+                "start_time": 1625204910,
+                "periods": [
+                    {
+                        "length_seconds": 2419200,
+                        "coins": "0aevmos",
+                    },
+                ],
+            },
+        },
+    ]
+
+    cli = evmos.cosmos_cli()
+    for tc in test_cases:
+        print("\nCase: {}".format(tc["name"]))
+        # create the vesting account
+        tx = cli.create_vesting_acc(
+            tc["funder"],
+            tc["address"],
+        )
+        tx = cli.sign_tx_json(tx, tc["address"], max_priority_price=0)
+
+        rsp = cli.broadcast_tx_json(tx, broadcast_mode="sync")
+        # assert tx returns OK code
+        assert rsp["code"] == 0
+
+        # wait tx to be committed
+        wait_for_new_blocks(cli, 2)
+
+        # funder funds the vesting account and defines the
+        # vesting and lockup schedules
+        with tempfile.NamedTemporaryFile("w") as lockup_file:
+            json.dump(tc["lockup"], lockup_file)
+            lockup_file.flush()
+
+            with tempfile.NamedTemporaryFile("w") as vesting_file:
+                json.dump(tc["vesting"], vesting_file)
+                vesting_file.flush()
+
+                # expect an error
+                try:
+                    tx = cli.fund_vesting_acc(
+                        tc["address"],
+                        tc["funder"],
+                        lockup_file.name,
+                        vesting_file.name,
+                    )
+                    raise Exception("This command should have failed")
+                except Exception as error:
+                    assert tc["exp_err"] in error.args[0]
+
+
+def test_unvested_token_delegation(evmos):
+    """
+    test vesting account cannot delegate unvested tokens
+    """
+    cli = evmos.cosmos_cli()
+    funder = eth_to_bech32(ADDRS["signer1"])
+    # add a new key that will be the vesting account
+    acc = cli.create_account("vesting_acc")
+    address = acc["address"]
+
+    # transfer some funds to pay for tx fees
+    # when creating the vesting account
+    tx = cli.transfer(
+        funder,
+        address,
+        f"{7000000000000000}{DEFAULT_DENOM}",
+        gas_prices=f"{cli.query_base_fee() + 100000}{DEFAULT_DENOM}",
+        generate_only=True,
+    )
+
+    tx = cli.sign_tx_json(tx, funder, max_priority_price=0)
+
+    rsp = cli.broadcast_tx_json(tx, broadcast_mode="sync")
+    assert rsp["code"] == 0, rsp["raw_log"]
+    # txhash = rsp["txhash"]
+
+    wait_for_new_blocks(cli, 2)
+
+    # create the vesting account
+    tx = cli.create_vesting_acc(funder, address)
+    tx = cli.sign_tx_json(tx, address, max_priority_price=0)
+    rsp = cli.broadcast_tx_json(tx, broadcast_mode="sync")
+
+    # assert tx returns OK code
+    assert rsp["code"] == 0
```

Comment:
in other cases the raw logs are printed if the assertion is not fulfillled, could also do that here
```suggestion
    assert rsp["code"] == 0, rsp["raw_log"]
```